server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2086,6 +2006,38 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
         return success;
     }
 
+    private void changeDiskOfferingForRootVolume(Long vmId, DiskOfferingVO newDiskOffering, Map<String, String> customParameters) throws ResourceAllocationException {
+
+        List<VolumeVO> vols = _volsDao.findReadyAndAllocatedRootVolumesByInstance(vmId);
+
+        for (final VolumeVO rootVolumeOfVm : vols) {
+            DiskOfferingVO currentRootDiskOffering = _diskOfferingDao.findById(rootVolumeOfVm.getDiskOfferingId());
+            HypervisorType hypervisorType = _volsDao.getHypervisorType(rootVolumeOfVm.getId());
+            if (HypervisorType.Simulator != hypervisorType) {
+                Long minIopsInNewDiskOffering = null;
+                Long maxIopsInNewDiskOffering = null;
+                boolean autoMigrate = false;
+                if (customParameters.containsKey(ApiConstants.MIN_IOPS)) {
```

Comment:
what if the new disk offering has min/max iops, and not specified through custom parameters?

Reply:
ScaleVM API has separate parameters (not from details/custom params) to pass min and max iops values. Internally in service layer during passing between methods these are added to custom parameters and used here.

Reply:
> 

So, only the min and max iops values passed through the API cmd are supported, right?